In [3]:

import numpy as np
import scipy as sp
import pandas as pd
from pandas import Series, DataFrame

import matplotlib.pyplot as plt
import matplotlib as mpl
import seaborn as sns
sns.set()
%matplotlib inline

%precision 3

'%.3f'

In [4]:
heart_disease = pd.read_csv('heart_disease.csv')

In [5]:
heart_disease.head()

age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  slope  \
0   63    1   3       145   233    1        0      150      0      2.3      0   
1   37    1   2       130   250    0        1      187      0      3.5      0   
2   41    0   1       130   204    0        0      172      0      1.4      2   
3   56    1   1       120   236    0        1      178      0      0.8      2   
4   57    0   0       120   354    0        1      163      1      0.6      2   

   ca  thal  target  
0   0     1       1  
1   0     2       1  
2   0     2       1  
3   0     2       1  
4   0     2       1

In [6]:
heart_disease.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 303 entries, 0 to 302
Data columns (total 14 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       303 non-null    int64  
 1   sex       303 non-null    int64  
 2   cp        303 non-null    int64  
 3   trestbps  303 non-null    int64  
 4   chol      303 non-null    int64  
 5   fbs       303 non-null    int64  
 6   restecg   303 non-null    int64  
 7   thalach   303 non-null    int64  
 8   exang     303 non-null    int64  
 9   oldpeak   303 non-null    float64
 10  slope     303 non-null    int64  
 11  ca        303 non-null    int64  
 12  thal      303 non-null    int64  
 13  target    303 non-null    int64  
dtypes: float64(1), int64(13)
memory usage: 33.3 KB


欄位說明：
1. age - age in years
2. sex -（1 =男性； 0 =女性）
3. cp - -胸痛型別（0: 典型的心絞痛：與胸痛有關，減少心臟的血液供應1: 非心絞痛：通常為食道痙攣（與心臟無關）2: 非心絞痛：通常為食道痙攣（與  心臟無關）3: 無症狀：胸痛未顯示疾病跡象）
4. trestbps - 靜息血壓（入院時以毫米汞柱為單位）高於130-140的情況通常令人擔憂
5. chol - 血清膽汁，mg / dl
    血清= LDL + HDL + 0.2 *甘油三酸酯
    200以上值得關注
6. fbs-（空腹血糖> 120 mg / dl）（1 =正確; 0 =錯誤）
    '> 126’mg / dL表示糖尿病
7. restecg --靜息心電圖結果
    0：沒什麼可注意的
    1：ST-T波異常
    範圍從輕度症狀到嚴重問題
    表示心跳不正常
    2:可能或確定的左心室肥大
    擴大心臟的主泵室
8. 丘腦-達到最大心率
9. exang - 運動誘發的心絞痛（1 =是; 0 =否）
10. oldpeak - 運動引起的相對於休息的ST抑鬱會在運動不健康的心臟時觀察心臟的壓力，這會增加壓力
11. slope - 最高運動ST段的坡度
    0: Upsloping: 鍛鍊可提高心率（罕見）
    1: Flatsloping: 變化很小（典型的健康心臟）
    2: Downslopins:心臟不健康的體徵
12. ca - 螢光顯色的主要血管數目（0-3）
    彩色血管意味著醫生可以看到血液通過
    血液運動越多越好（無凝塊）
13. thal - al-lium應力結果
    1,3:正常
    6: 修復缺陷：曾經是缺陷，但是現在可以了
    7: 可逆缺陷：鍛鍊時沒有適當的血液運動
14. target - 是否患有疾病（1 =是，0 =否）（=預測屬性）

In [7]:
#檢查是否有空值
heart_disease.isna().sum()

age         0
sex         0
cp          0
trestbps    0
chol        0
fbs         0
restecg     0
thalach     0
exang       0
oldpeak     0
slope       0
ca          0
thal        0
target      0
dtype: int64

In [8]:
heart_disease['sex'].head()

0    1
1    1
2    0
3    1
4    0
Name: sex, dtype: int64

In [12]:
#男性和女性的trestbps靜息血壓平均值，最大值，最小值
heart_disease.groupby('sex')['trestbps'].mean()

sex
0    133.083333
1    130.946860
Name: trestbps, dtype: float64

In [14]:
heart_disease.groupby('sex')['trestbps'].max()

sex
0    200
1    192
Name: trestbps, dtype: int64

In [13]:
heart_disease.groupby('sex')['trestbps'].min()

sex
0    94
1    94
Name: trestbps, dtype: int64

In [19]:
#男性和女性的chol血清膽汁平均值，最大值，最小值
#血清= LDL + HDL + 0.2 *甘油三酸酯
#200以上值得關注
heart_disease.groupby('sex')['chol'].mean()

sex
0    261.302083
1    239.289855
Name: chol, dtype: float64

In [20]:
heart_disease.groupby('sex')['chol'].max()

sex
0    564
1    353
Name: chol, dtype: int64

In [21]:
heart_disease.groupby('sex')['chol'].min()

sex
0    141
1    126
Name: chol, dtype: int64